app/controllers/announcements_controller.rb

```diff
@@ -0,0 +1,71 @@
+class AnnouncementsController < ApplicationController
+  before_action :set_announcement, only: [:edit, :update, :destroy]
+  before_action :new_announcement, only: [:new, :create, :index]
+
+  def show
+    redirect_to action: "index"
+  end
+
+  def index
+    @announcement = Announcement.joins(:user).active_announcements.not_expired.order(reveal_date: :desc)
+  end
+
+  def new
+    @announcement.reveal_date = Date.today
+    @announcement.expires_at = Date.today + 1.months
+  end
+
+  def create
+    @announcement = Announcement.new(announcement_params)
+    @announcement.user_id = current_user.id
+
+    if @announcement.save
+      redirect_to action: "index"
```

Comment:
I don't see exactly what's wrong here. Can you give an example of what you mean?

It should be redirecting to the index page on success and on a failure it re renders the new announcement page with an error.